general/scans/scans.py

```diff
@@ -300,6 +311,210 @@ def __repr__(self):
+        import warnings
+        warnings.simplefilter("ignore", UserWarning)
+
+        detector = self._normalise_detector(detector)
+        fig, axis = plt.subplots()
+        plt.show()
+
+        xs = []
+        ys = ListOfMonoids()
+
+        action_remainder = None
+
+        try:
+            with open(self.defaults.log_file(), "w") as logfile, \
+                    detector(self, save=save, **kwargs) as detect:
+
+                for move in self:
+                    # Set initial motor position to correct value.
+                    if abs(self.motion() - move.start) > self.motion.tolerance:
+                        self.motion(move.start)
+                        while abs(self.motion() - move.start) > \
+                                self.motion.tolerance:
+                            time.sleep(0.1)
+
+                    with temporarily_change_motor_speed(self.motion,
+                                                        move.speed):
+
+                        self.motion(move.stop)
+
+                        while abs(self.motion() - move.stop) > \
+                                self.motion.tolerance:
+
+                            position, value = self.motion(), Exact(
+                                detect(**just_times(kwargs)))
+
+                            xs.append(position)
+                            ys.append(value)
+
+                            logfile.write("{}\t{}\n".format(xs[-1],
+                                                            str(ys[-1])))
+                            axis.clear()
+
+                            if isinstance(self.min(), tuple):
+                                rng = [1.05 * self.min()[0] -
+                                       0.05 * self.max()[0],
+
+                                       1.05 * self.max()[0] -
+                                       0.05 * self.min()[0]]
+                            else:
+                                rng = [1.05 * self.min() - 0.05 * self.max(),
+                                       1.05 * self.max() - 0.05 * self.min()]
+                            axis.set_xlim(rng[0], rng[1])
+                            rng = _plot_range(ys)
+                            axis.set_ylim(rng[0], rng[1])
+                            ys.plot(axis, xs)
+                            if action:
+                                action_remainder = action(xs, ys, axis)
+
+                            plt.draw()
+
+                            # If we plot in a tight loop, matplotlib can't keep
+                            # up. Taking data at 5Hz during the move seems the
+                            # right balance of "continuous" and "pragmatic"
+                            #
+                            # Note: a galil's MAX update frequency is 40ms so
+                            # there is no benefit in making this number smaller
+                            # than 0.04
+                            time.sleep(0.2)
```

Comment:
Could this be an optional argument with a default value so the user can choose to scan continuously at different frequencies